src/lib.rs

```diff
@@ -146,75 +146,32 @@ mod quote;
 
 mod error;
 mod iter;
-mod parse;
 
-use crate::error::{compile_error, Error};
-use crate::iter::Iter;
-use crate::parse::{
-    parse_define_args, parse_enum_hack, parse_export_args, parse_fake_call_site, parse_input,
-};
-use proc_macro::{Ident, Punct, Spacing, Span, TokenStream, TokenTree};
-use std::fmt::Write;
+#[cfg_attr(not(fn_like_proc_macro), path = "fallback/mod.rs")]
+#[cfg_attr(fn_like_proc_macro, path = "compiler.rs")]
+mod imp;
+use crate::imp::*;
```

Comment:
I had to use `cfg` for more than half of `lib.rs` and `perse.rs`'s items, so I separated the files that have code that needs `cfg`.